Realm/Realm/Realm.cs

```diff
@@ -626,20 +626,16 @@ public RealmObject Add(RealmObject obj, bool update = false)
             return obj;
         }
 
-        internal void ManageEmbedded(EmbeddedObject obj, ObjectHandle handle)
+        internal void ManageEmbedded(IEmbeddedObject obj, ObjectHandle handle)
         {
             var objectType = obj.GetType();
             var objectName = objectType.GetMappedOrOriginalName();
             Argument.Ensure(Metadata.TryGetValue(objectName, out var metadata), $"The class {objectType.Name} is not in the limited set of classes for this realm", nameof(obj));
 
-            obj.SetOwner(this, handle, metadata);
-
-            // If an object is newly created, we don't need to invoke setters of properties with default values.
-            metadata.Helper.CopyToRealm(obj, update: false, skipDefaults: true);
-            obj.OnManaged();
+            obj.SetManagedAccessor(ManagedAccessor.Create(this, handle, metadata), () => { metadata.Helper.CopyToRealm(obj, false, true); });
```

Comment:
Rather than passing a lambda here, should we add the relevant arguments to `SetManagedAccessor` and have it call `metadata.Helper.CopyToRealm` internally?

Reply:
`SetManagedAccessor` is defined on `IRealmObjectBase`, so it will be source generated. If we do it inside the method, then we would need to make `metadata.Helper.CopyToRealm` a public method.

Reply:
CopyToRealm will probably need to be source generated as well - this is currently generated by the weaver.